Applications/Spire/Source/Ui/ListValueModel.cpp

```diff
@@ -0,0 +1,61 @@
+#include "Spire/Ui/ListValueModel.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+ListValueModel::ListValueModel(std::shared_ptr<ListModel> source, int index)
+    : m_source(std::move(source)),
+      m_index(index) {
+  m_operation_connection = m_source->connect_operation_signal(
+    std::bind_front(&ListValueModel::on_operation, this));
+}
+
+QValidator::State ListValueModel::get_state() const {
+  if(m_index < 0 || m_index >= m_source->get_size()) {
+    return QValidator::State::Invalid;
+  } else {
+    return QValidator::State::Acceptable;
+  }
+}
+
+const ListValueModel::Type& ListValueModel::get_current() const {
+  if(get_state() == QValidator::State::Acceptable) {
+    return m_source->at(m_index);
+  } else {
```

Comment:
This else statement and the else statement in get_state() can be removed, because when the if statement is false then the body of the else statement will always be executed.